File: 704.binary-search.cs
/*
 * @lc app=leetcode id=704 lang=csharp
 *
 * [704] Binary Search
 */

// @lc code=start
public class Solution {
    public int Search(int[] nums, int target) {
        int low = 0;
        int high = nums.Length - 1;
        while(low < high){
            int mid = Mid(low,high);
            if(target > nums[mid])
                low = mid + 1;
            else 
                high = mid;
        }
        return target == nums[low] ? low : -1;  
    }

    public int Mid(int low, int high) => low + (high-low) / 2;
}
// @lc code=end

File: 704.binary-search.js
/*
 * @lc app=leetcode id=704 lang=javascript
 *
 * [704] Binary Search
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function (nums, target) {
  var min = 0;
  var max = nums.length - 1;
  while (min < max) {
    var mid = Math.floor(min + (max - min) / 2);
    if (nums[mid] < target) min = mid + 1;
    else max = mid;
  }
  return nums[min] == target ? min : -1;
};
console.log(search([2, 5], 5));
// @lc code=end
